### PR TITLE
add view profile and edit profile pages

### PR DESCRIPTION
**Pull request by teenoh**

### CODE REVIEW

File: components/profile/components/EditProfile/index.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    locationLabel,
+    profilePhotoLabel,
+    profilePhotoPreviewAlt,
+    uploadLabel,
+} = EDIT_PROFILE_FIELDS;
+
+class EditProfile extends Component {
+    state = { 
+        fileList: [], location: null, locationLoading: false, 
+        previewImage: '', previewVisible: false,
```
Comment (enyeInc) [inline components/profile/components/EditProfile/index.js]: if keys can not all fit on one line, put them one there own line

File: components/profile/components/EditProfile/index.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+class EditProfile extends Component {
+    state = { 
+        fileList: [], location: null, locationLoading: false, 
+        previewImage: '', previewVisible: false, 
+    }
+
+    handleImagePreview = file => {
+        this.setState({
+            previewImage: file.url || file.thumbUrl,
```
Comment (enyeInc) [inline components/profile/components/EditProfile/index.js]: destructure `file`

File: components/profile/components/EditProfile/index.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                        <TextArea className={`${w80}`} rows={4} />
+                    </div>
+                </Col>
+                <Col span={24}>
+                    <Button type={BUTTON_PRIMARY}>{EDIT_PROFILE_TITLE}</Button>
+                </Col>
+            </Row>
+        );
+    }
+}
```
Comment (enyeInc) [inline components/profile/components/EditProfile/index.js:152]: This file has alot of repeated code, can you refactor it by mapping over an array of objects to create these code chunks. This is very messy as it stands, please also use a similar pattern in other parts of your code base

Exampl:
```
[{}, {}, {}].map(item => {
    const { className, label, data } = item;
    // use data to determine that
    const element = generateElement(data); // you also don't have to call it element
    
    return (
        <div className={className}>
            <h4>{label}</h4>
            { element }
        </div>
    )
});
```

File: pages/admin/profile/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        if (action === 'view') {
+            return this.setState({ action: 'edit' });
+        }
+
+        return this.setState({ action: 'view' });
+    }
+
+    render() {
+        const { action } = this.state;
+        return (
```
Comment (enyeInc) [inline pages/admin/profile/index.js:21]: define a variable `const isViewAction = action === 'view'` and use that in the render function

File: pages/admin/profile/index.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            <Layout selectedMenuItem="profile">
+                <Row>
+                    <Col span={24} className='mb-1'>
+
```
Comment (enyeInc) [inline pages/admin/profile/index.js]: nit: remove unused line

File: components/profile/components/ViewProfile/index.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+                </Row>
+            </Col>
+        </Row>
+    );};
+
+ViewProfile.defaultProps = {
+    user: DEFAULT_USER_PROPS,
+};
+
+export default ViewProfile;
```
Comment (enyeInc) [inline components/profile/components/ViewProfile/index.js:100]: go through this file and use the map pattern from above

**Comment by enyeInc** [comment]
@teenoh I trust that the code functionally works, you just need to clean up your code a bit. If you are confused or need help in doing this, we can sit down for 15-20 mins

File: components/profile/components/EditProfile/index.js (hunk 1 above)
Comment (teenoh) [reply components/profile/components/EditProfile/index.js]: okay. Noted

File: components/profile/components/EditProfile/index.js (hunk 3 above)
Comment (teenoh) [reply components/profile/components/EditProfile/index.js:152]: I thought so too. That generateElement is a great idea. Thanks

**Review by enyeInc — APPROVED** [review]
Code has some conventions that needs ironing, but we will get there. For now this looks good to be merged

**Review by tosinamuda — APPROVED** [review]

**Pull request merged**